File: CA 1/Practice 1/discarded_cards.py
def discarded_cards(n):
    q1 = [i for i in range(1, n+1)]
    q2 = []
    for _ in range(n):
        if len(q1) <= 1: break
        q2.append(q1.pop(0))
        q1.append(q1.pop(0))
    return q2, q1


while True:
    n = int(input())
    if n == 0: break
    cards = discarded_cards(n)
    print(cards)
    print(f"Discarded cards: {', '.join(map(str, cards[0]))}")
    print(f'Remaining card: {cards[1][0]}')

File: CA 1/Practice 1/discarded cards.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n{}, x{};
    queue<int> q;
    while (1)
    {
        cin >> n;
        if (!n)
            break;

        
        for (int i{1}; i <= n; i++)
            q.push(i);
        cout << "Discarded cards:";
        while (q.size() > 1)
        {
            cout << " " << q.front();
            q.pop();
            x = q.front();
            q.pop();
            if (!q.empty())
                cout << ",";
            q.push(x);
        }
        cout << "\nRemaining card: " << q.front() << "\n";
        q.pop();
    }

    return 0;
}
